Move FreeBSD builds to DncEng Pool provider builds

MattGal opened the pull request:
From chatting with @wfurt it seems FreeBSD's [very-large docker images are filling up the disk](https://dev.azure.com/dnceng/public/_build/results?buildId=618060&view=logs&jobId=dd0b2759-5918-5f16-563d-5e53487f9810&j=dd0b2759-5918-5f16-563d-5e53487f9810&t=4d22f787-eee9-50d3-a257-5cb17b9a6e23).

MattGal committed:
Move FreeBSD builds to DncEng Pool provider builds

wfurt commented:
```
mcr.microsoft.com/dotnet-buildtools/prereqs         ubuntu-18.04-cross-freebsd-11-20200422191843-a84b0d2     82e7c04db157        5 days ago          2.9GB
mcr.microsoft.com/dotnet-buildtools/prereqs         ubuntu-16.04-cross-arm64-20200413125008-cfdd435          74a1c845cec7        2 weeks ago         3.44GB
mcr.microsoft.com/dotnet-buildtools/prereqs         ubuntu-18.04-cross-freebsd-11-20200407092345-a84b0d2     114c2be50e53        2 weeks ago         2.9GB
```

It is interesting that it claims to be smaller than the arm64.

MattGal commented:
> ```
> mcr.microsoft.com/dotnet-buildtools/prereqs         ubuntu-18.04-cross-freebsd-11-20200422191843-a84b0d2     82e7c04db157        5 days ago          2.9GB
> mcr.microsoft.com/dotnet-buildtools/prereqs         ubuntu-16.04-cross-arm64-20200413125008-cfdd435          74a1c845cec7        2 weeks ago         3.44GB
> mcr.microsoft.com/dotnet-buildtools/prereqs         ubuntu-18.04-cross-freebsd-11-20200407092345-a84b0d2     114c2be50e53        2 weeks ago         2.9GB
> ```
> 
> It is interesting that it claims to be smaller than the arm64.

sure but why are they both so abnormally big?

wfurt commented:
oh. we already pick same pool for linux.

wfurt commented:
> sure but why are they both so abnormally big?

not sure what 'abnormally' means. It has OS + LLVM is ~ 1G + QEMU(static) + and other packages for binary manipulation. 

now the rootfs as yet another copy of OS - at least enough to have libraries and packages we need for build. 

That is still tiny comparing to OSX or Windows.

safern commented on eng/pipelines/common/xplat-setup.yml:
```diff
@@ -97,6 +97,16 @@ jobs:
           queue: BuildPool.Ubuntu.1604.Amd64
 
+        # Public FreeBSD Build Pool
+        ${{ if and(eq(parameters.osGroup, 'FreeBSD_x64'), eq(variables['System.TeamProject'], 'public')) }}:
```
I don’t think this is going to differ ever from the Linux pools as this is now contenirized. Could we merge the two conditions instead?

MattGal replied:
Sure.  I wasn't clear on the future of how this works as we might want to have build agents running natively on BSD when it becomes possible.

MattGal committed:
PR feedback

MattGal committed:
Update xplat-setup.yml

From reading @wfurt 's commit it looks like osGroup is actually FreeBSD here.  FreeBSD_x64 seems to be ignored.
